jwst/stpipe/step.py

```diff
@@ -1264,6 +1264,30 @@ def get_pars_model(step, full_spec=True):
 
         return pars_model
 
+    def update(self, attributes):
+        """Update step attributes
+
+        Parameters
+        ----------
+        attributes : dict
+            Attributes to update. If a value is another dict,
+            recursively update those attributes.
+
+        Notes
+        -----
+        `attributes` is presumed to have been produced by the
+        `Step.get_pars` method. As such, the "steps" key is treated
+        special in that it is a dict whose keys are the steps assigned
+        directly as attributes to the current step. This is standard
+        practice for `Pipeline`-based steps.
```

Comment:
It is not at all clear what this method does.  Also, it looks a bit dangerous.  Wouldn't it be better to have `Step.get_pars` have a post-hook that does the update, so the functionality here is tied to `Step.get_pars`?  Especially if the assumption is the dict is produced by `Step.get_pars`.

Reply:
Offline discussion clarified the need. Actions are:

- More commentary
- Protect against updating attributes that are not already part of the relevant steps.
- Rename to a more specific function, such as `update_step_parameters`